dotnet/src/SemanticKernel/SkillDefinition/SKFunction.cs

```diff
@@ -263,20 +269,11 @@ public override string ToString()
     public string ToString(bool writeIndented)
         => JsonSerializer.Serialize(this, options: writeIndented ? s_toStringIndentedSerialization : s_toStringStandardSerialization);
 
-    /// <summary>
-    /// Finalizer.
-    /// </summary>
-    ~SKFunction()
-    {
-        this.ReleaseUnmanagedResources();
-    }
-
     #region private
 
     private static readonly JsonSerializerOptions s_toStringStandardSerialization = new();
     private static readonly JsonSerializerOptions s_toStringIndentedSerialization = new() { WriteIndented = true };
-    private readonly DelegateTypes _delegateType;
-    private readonly Delegate _function;
+    private readonly Func<ITextCompletion?, CompleteRequestSettings?, SKContext, Task<SKContext>> _function;
```

Comment:
I know we're already kind of tied to ITextCompletion, but wondering what this might look like when|if we support IImageGeneration or other IText* variants, as SKFunctions.

Reply:
This includes ITextCompletion just to be able to unify the semantic and native functions into a single delegate; the native functions ignore the first argument.  This is all a private implementation detail and can evolve to whatever the semantic invocation needs.